WcaOnRails/app/webpacker/components/RolesTab/index.jsx

```diff
@@ -6,77 +6,136 @@ import {
+      return false;
+    },
+    [data],
   );
-}
 
-export default function RolesTab({ userId }) {
-  const {
-    data, sync, loading, error,
-  } = useLoadedData(roleListUrl(userId));
+  React.useEffect(() => {
+    if (!loggedInUserFetch.loading) {
+      save(roleListUrl(loggedInUserFetch.data.id), {}, (roles) => {
+        setCurrentUserRoles(roles.activeRoles.reduce((roleMap, role) => ({
+          ...roleMap,
+          [role.group.id]: role,
+        }), {}));
+      }, { method: 'GET', body: null }, setError);
+    }
+  }, [loggedInUserFetch.data, loggedInUserFetch.loading, save]);
+
+  const canEditRole = (role) => role.group.group_type === 'delegate_regions' && (
+    !!currentUserRoles.admin || currentUserRoles[role.group.id].status === 'senior_delegate'
+  );
 
-  if (loading) return <Loading />;
-  if (error) return <Errored />;
+  const canEditTeamOfRole = (role) => role.group.group_type !== 'delegate_regions' && (
+    !!currentUserRoles.admin || currentUserRoles[role.group.name]?.status === 'leader'
+  );
+
+  if (
+    roleListFetch.loading || loggedInUserFetch.loading || saving || currentUserRoles === null
+  ) return <Loading />;
+  if (roleListFetch.error || loggedInUserFetch.error || error) return <Errored />;
 
-  return (data.activeRoles.length > 0
-    ? (
-      <>
-        <Header>Active Roles</Header>
-        <List>
-          <List.Item>
-            <RoleFormModal
-              trigger={(
-                <Button.Group basic vertical>
-                  <Button>Delegate</Button>
-                </Button.Group>
-          )}
-              userId={userId}
-              isActiveRole
-              sync={sync}
-            />
+  return (
+    <>
+      {activeNonHiddenRoles.length > 0
+        ? (
+          <>
+            <Header>Active Roles</Header>
+            <List divided relaxed>
+              {activeNonHiddenRoles.map((role) => (
+                <List.Item>
+                  <List.Icon
+                    name="edit"
+                    size="large"
+                    verticalAlign="middle"
+                    link
+                    disabled={!canEditRole(role)}
+                    onClick={() => setOpen(true)}
+                  />
+                  <List.Content>
+                    {canEditTeamOfRole(role) && (
+                    <List.Header as="a" href={`${teamUrl(role.group.id)}/edit`}>
+                      {`${I18n.t(`enums.user.role_status.${role.status}`)}, ${role.group.name}`}
+                    </List.Header>
+                    )}
+                    {!canEditTeamOfRole(role) && (
+                    <List.Header>
+                      {`${I18n.t(`enums.user.role_status.${role.status}`)}, ${role.group.name}`}
+                    </List.Header>
+                    )}
+                    {!!role.start_date && (
+                    <List.Description>
+                      {`Since ${role.start_date}`}
+                    </List.Description>
+                    )}
```

Comment:
The indentation seems off here, for example. The `List` should be indented more.

Reply:
Oh okay, somehow eslint formatter didn't fix that. It didn't throw warning/error as well in both correct indentation and wrong indentation. I've anyway added more indentation now.